processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go

```diff
@@ -22,24 +22,33 @@ import (
 type spanCount struct {
 	logger   *zap.Logger
 	minSpans int32
+	maxSpans int32
 }
 
 var _ PolicyEvaluator = (*spanCount)(nil)
 
 // NewSpanCount creates a policy evaluator sampling traces with more than one span per trace
-func NewSpanCount(logger *zap.Logger, minSpans int32) PolicyEvaluator {
+func NewSpanCount(logger *zap.Logger, minSpans, maxSpans int32) PolicyEvaluator {
 	return &spanCount{
 		logger:   logger,
 		minSpans: minSpans,
+		maxSpans: maxSpans,
 	}
 }
 
 // Evaluate looks at the trace data and returns a corresponding SamplingDecision.
 func (c *spanCount) Evaluate(_ pcommon.TraceID, traceData *TraceData) (Decision, error) {
 	c.logger.Debug("Evaluating spans counts in filter")
 
-	if int(traceData.SpanCount.Load()) >= int(c.minSpans) {
+	spanCount := traceData.SpanCount.Load()
+	switch {
+	case c.maxSpans == 0 && int(spanCount) >= int(c.minSpans):
```

Comment:
You are potentially casting the spanCount to int multiple times, and there's never a situation you are not casting it. Why not just cast it on line 43 directly?

Reply:
Yes, it would be better